src/main/java/dao/impl/JDBC/OrderDaoJDBC.java

```diff
@@ -0,0 +1,146 @@
+import utils.ConnectionJDBC;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class OrderDaoJDBC implements OrderDao {
+
+    private Logger logger = Logger.getLogger(OrderDaoJDBC.class);
+
+    @Override
+    public void addOrder(Order order) {
+        try (Connection connection = ConnectionJDBC.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "INSERT INTO order_test (address, payment, code, products, userid) "
+                            + "VALUES (?,?,?,?,?)");
+            statement.setString(1, order.getAddress());
+            statement.setString(2, order.getPayment());
+            statement.setInt(3, order.getCode().getCodeValue());
+            statement.setString(4, order.getListOfProductsString());
+            statement.setLong(5, order.getCode().getUser().getId());
+            statement.execute();
+            logger.info(String.format("Added order for userID='%s'",
+                    order.getCode().getUser().getId()));
+        } catch (SQLException e) {
+            logger.error("Failed adding order", e);
+        }
+    }
+
+    @Override
+    public Optional<Order> getById(long id) {
+        try (Connection connection = ConnectionJDBC.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT order_test.id, address, payment, code, products, userid, email, password, role"
+                            + " FROM order_test INNER JOIN users_test ON userid = users_test.id"
+                            + " WHERE order_test.id=?");
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            return getOrderFromResultset(resultSet);
+        } catch (SQLException e) {
+            logger.error(String.format("Failed getting order with id='%s'", id), e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<Order> getByCode(Code code) {
+        try (Connection connection = ConnectionJDBC.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT order_test.id, address, payment, code, products, userid, email, password, role "
+                            + "FROM order_test INNER JOIN users_test "
+                            + "ON userid = users_test.id WHERE code=? AND userid=?");
+            statement.setString(1, Integer.toString(code.getCodeValue()));
+            statement.setLong(2, code.getUser().getId());
+            ResultSet resultSet = statement.executeQuery();
+            return getOrderFromResultset(resultSet);
+        } catch (SQLException e) {
+            logger.error(String.format("Failed getting order where userid='%s'",
```

Comment:
if we want to use integer we can write:
"Failed getting order where userid = %d"